### PR TITLE
elf: reduce noise in the developer debug logs

### PR DESCRIPTION
**Pull request by cjp256**
Remove logs for:
- skipping unrelated links
- ldd actions (the parsed output is still logged)

Signed-off-by: Chris Patterson <chris.patterson@canonical.com>

- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

File: snapcraft/internal/elf.py [hunk 1]
```diff
@@ -62,7 +60,6 @@ def ldd(path: str, ld_library_paths: List[str]) -> Dict[str, str]:
 
     env = os.environ.copy()
     env["LD_LIBRARY_PATH"] = ":".join(ld_library_paths)
-    logger.debug(f"invoking ldd with ld library paths: {ld_library_paths!r}")
```
Comment (sergiusens) [inline snapcraft/internal/elf.py:65]: We had many issues in the past when calling ldd with variations of LD_LIBRARY_PATH

File: snapcraft/internal/elf.py [hunk 2]
```diff
@@ -74,7 +71,6 @@ def ldd(path: str, ld_library_paths: List[str]) -> Dict[str, str]:
         ldd_lines = (
             subprocess.check_output(["ldd", path], env=env).decode().splitlines()
         )
-        logger.debug(f"ldd output:\n{ldd_lines}")
```
Comment (sergiusens) [inline snapcraft/internal/elf.py:77]: If removing this, at least log things that don't match on https://github.com/snapcore/snapcraft/pull/3287/files#diff-f7cf307285f042f35434bbe680606c77R91 but skip logging for the usual suspects, we might fall into a parsing logic error and this really helped in the past.

**Review by sergiusens — CHANGES_REQUESTED** [review]
Looks good, just a few observations

File: snapcraft/internal/elf.py (hunk 2 above)
Comment (cjp256) [reply snapcraft/internal/elf.py:77]: Warn about failed matches if unexpected.

**Comment by sergiusens** [comment]
Closing stale PR

**Pull request closed without merging**